src/main/scala/diplomacy/LazyModule.scala

```diff
@@ -4,56 +4,100 @@ package freechips.rocketchip.diplomacy
+import scala.collection.immutable.{SortedMap, ListMap}
 import scala.util.matching._
 
+/** [[LazyModule]] uses Scala's lazy evaluation property to 
+  *  split hardware generation into two phases.
+  * - Phase 1 (non-lazy) states hardware intent and constraints:
+  * -- [[LazyModule]] and [[BaseNode]] instantiation.
+  * -- [[BaseNode]] binding.
+  * - Phase 2 (lazy) generates hardware:
+  * -- Parameter negotiation.
+  * -- [[AutoBundle]] resolution.
+  * -- [[LazyModuleImpLike]] module generation.
+  * */
 abstract class LazyModule()(implicit val p: Parameters)
 {
+  /** Contains sub-[[LazyModule]]s; can be accessed by [[getChildren]]. */
   protected[diplomacy] var children = List[LazyModule]()
+  /** Contains the [[BaseNode]]s instantiated within this instance. */
   protected[diplomacy] var nodes = List[BaseNode]()
+  /** Stores [[SourceInfo]] of this instance.
+    * The companion object factory method will set this to the correct value.
+    */
   protected[diplomacy] var info: SourceInfo = UnlocatableSourceInfo
-  protected[diplomacy] val parent = LazyModule.scope
+  /** Parent of this LazyModule. If this instance is at the top of the hierarchy, this will be [[None]]. */
+  protected[diplomacy] val parent: Option[LazyModule] = LazyModule.scope
 
-  // code snippets from 'InModuleBody' injection
+  /** Code snippets from [[InModuleBody]] injection. */
   protected[diplomacy] var inModuleBody = List[() => Unit]()
 
+  /** Chain of ancestor LazyModules, starting with [[parent]] on left. */
   def parents: Seq[LazyModule] = parent match {
     case None => Nil
     case Some(x) => x +: x.parents
   }
 
+  // [[LazyModule.scope]] stack push.
   LazyModule.scope = Some(this)
+  // Prepend this instance to [[parent]]'s list of children.
   parent.foreach(p => p.children = this :: p.children)
 
-  // suggestedName accumulates Some(names), taking the final one. Nones are ignored.
+  /** Accumulates Some(names), taking the final one. `None`s are ignored. */
   private var suggestedNameVar: Option[String] = None
+
+  /** Set instance name of [[LazyModuleImpLike]] module.
+    *
+    * @param x suggested name to [[LazyModuleImpLike]] module.
+    * */
   def suggestName(x: String): this.type = suggestName(Some(x))
   def suggestName(x: Option[String]): this.type = {
     x.foreach { n => suggestedNameVar = Some(n) }
     this
   }
 
+  /** Helper function of [[className]]. This searches upwards in the class hierarchy for the first class name that was not an anonymous scala class.*/
   private def findClassName(c: Class[_]): String = {
     val n = c.getName.split('.').last
     if (n.contains('$')) findClassName(c.getSuperclass) else n
   }
 
-  lazy val className = findClassName(getClass)
-  lazy val suggestedName = suggestedNameVar.getOrElse(className)
-  lazy val desiredName = className // + hashcode?
+  /** Get the class name of this instance. */
+  lazy val className: String = findClassName(getClass)
+  /** Set instance name. */
```

Comment:
How about "Set suggested instance name"?

Reply:
No, the purpose of my suggestion here is the same as the one immediately above it, which is that members should have "noun-like" descriptions and functions should have "verb-like" descriptions. It's odd to describe a `val` as something that "performs an action" because a `val` is just something that _is_.

Reply:
And to illustrate this, here's a screenshot of what this looks like in HTML form:

<img width="956" alt="Screen Shot 2020-03-11 at 11 23 09 PM" src="https://user-images.githubusercontent.com/1002748/76493184-47228680-63ef-11ea-9807-caa6e5df955f.png">

Reply:
ok, i am fine either way

Reply:
OK, since I have an opinion on this, I will make these changes.